packages/sirius-web/frontend/sirius-web-application/src/modals/duplicate-object/DuplicateObjectModal.tsx

```diff
@@ -0,0 +1,270 @@
+ * https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *     Obeo - initial API and implementation
+ *******************************************************************************/
+import { Selection, SelectionContext } from '@eclipse-sirius/sirius-components-core';
+import { TreeView, TreeItemActionProps } from '@eclipse-sirius/sirius-components-trees';
+import UnfoldMoreIcon from '@mui/icons-material/UnfoldMore';
+import Checkbox from '@mui/material/Checkbox';
+import Button from '@mui/material/Button';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import FormControlLabel from '@mui/material/FormControlLabel';
+import IconButton from '@mui/material/IconButton';
+import MenuItem from '@mui/material/MenuItem';
+import Select, { SelectChangeEvent } from '@mui/material/Select';
+import { useEffect, useState } from 'react';
+import { makeStyles } from 'tss-react/mui';
+import { DuplicateObjectModalProps, DuplicateObjectModalState } from './DuplicateObjectModal.types';
+import { FilterBar } from './FilterBar';
+import { useContainmentFeatureNames } from './useContainmentFeatureNames';
+import { useDuplicateObject } from './useDuplicateObject';
+import { useDuplicateDialogTreeSubscription } from './useDuplicateDialogTreeSubscription';
+
+const useDuplicateObjectModalStyles = makeStyles()((theme) => ({
+  form: {
+    display: 'flex',
+    flexDirection: 'column',
+    '& > *': {
+      marginBottom: theme.spacing(1),
+    },
+  },
+  title: {
+    opacity: 0.6,
+    fontSize: theme.typography.caption.fontSize,
+  },
+  borderStyle: {
+    border: '1px solid',
+    borderColor: theme.palette.grey[500],
+    height: 300,
+    overflow: 'auto',
+  },
+}));
+
+export const DuplicateObjectModal = ({
+  editingContextId,
+  objectToDuplicateId,
+  objectToDuplicateKind,
+  onObjectDuplicated,
+  onClose,
+}: DuplicateObjectModalProps) => {
+  const { classes } = useDuplicateObjectModalStyles();
+
+  const [state, setState] = useState<DuplicateObjectModalState>({
+    containerSelection: { entries: [] },
+    containmentFeatureNames: [],
+    containmentFeatureName: '',
+    duplicateContent: true,
+    copyOutgoingReferences: true,
+    updateIncomingReferences: false,
+    filterBarText: '',
+    expanded: [],
+    maxDepth: 1,
+  });
+
+  const { getContainmentFeatureNames, containmentFeatureNames } = useContainmentFeatureNames();
+
+  useEffect(() => {
+    if (state.containerSelection.entries[0]) {
+      getContainmentFeatureNames({
+        variables: {
+          editingContextId,
+          containerId: state.containerSelection.entries[0].id,
+          containedObjectId: objectToDuplicateId,
+        },
+      });
+    }
+  }, [state.containerSelection, objectToDuplicateId]);
```

Comment:
You don't need this to be based on useLazyQuery, you just need to expose skip as a property in `useContainmentFeatureNames`